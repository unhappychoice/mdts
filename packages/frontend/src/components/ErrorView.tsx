import React from 'react';
import { Box, Typography } from '@mui/material';
import { ReportProblemOutlined } from '@mui/icons-material';

interface ErrorViewProps {
  error: string | null;
}

const ErrorView: React.FC<ErrorViewProps> = ({ error }) => {
  if (!error) {
    return null;
  }

  const message = error.includes('404') ? 'Not Found' : `Error: ${error}`;

  return (
    <Box
      sx={{
        flexGrow: 1,
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        height: '100%',
        p: 4,
      }}
    >
      <ReportProblemOutlined sx={{ fontSize: 60, mb: 2 }} color="action" />
      <Typography variant="h5" color="text.secondary">
        {message}
      </Typography>
    </Box>
  );
};

export default ErrorView;
