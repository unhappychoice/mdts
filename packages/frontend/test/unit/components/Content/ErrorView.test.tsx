import { render, screen } from '@testing-library/react';
import React from 'react';
import ErrorView from '../../../../src/components/ErrorView';

describe('ErrorView', () => {
  test('renders nothing when error is null', () => {
    const { container } = render(<ErrorView error={null} />);
    expect(container).toBeEmptyDOMElement();
  });

  test('renders nothing when error is an empty string', () => {
    const { container } = render(<ErrorView error="" />);
    expect(container).toBeEmptyDOMElement();
  });

  test('renders error message correctly', () => {
    render(<ErrorView error="Something went wrong" />);
    expect(screen.getByText('Error: Something went wrong')).toBeInTheDocument();
  });

  test('renders "Not Found" for 404 errors', () => {
    render(<ErrorView error="HTTP error! status: 404" />);
    expect(screen.getByText('Not Found')).toBeInTheDocument();
  });

  test('renders snapshot correctly with an error', () => {
    const { asFragment } = render(<ErrorView error="Test error message" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders snapshot correctly with 404 error', () => {
    const { asFragment } = render(<ErrorView error="HTTP error! status: 404" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
