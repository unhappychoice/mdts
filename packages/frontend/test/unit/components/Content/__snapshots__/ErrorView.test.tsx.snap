// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ErrorView renders snapshot correctly with 404 error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1f23zer"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-17g5et6-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99 19.53 19H4.47zM12 2 1 21h22zm1 14h-2v2h2zm0-6h-2v4h2z"
      />
    </svg>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1sojvhz-MuiTypography-root"
    >
      Not Found
    </h5>
  </div>
</DocumentFragment>
`;

exports[`ErrorView renders snapshot correctly with an error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1f23zer"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-17g5et6-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99 19.53 19H4.47zM12 2 1 21h22zm1 14h-2v2h2zm0-6h-2v4h2z"
      />
    </svg>
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1sojvhz-MuiTypography-root"
    >
      Error: Test error message
    </h5>
  </div>
</DocumentFragment>
`;
